server/src/test/java/io/spine/server/bc/given/BoundedContextTestEnv.java

```diff
@@ -0,0 +1,236 @@
+        }
+
+        @Apply
+        private void event(ProjectStarted event) {
+            getBuilder()
+                    .setId(event.getProjectId())
+                    .setStatus(Project.Status.STARTED)
+                    .build();
+
+            isProjectStartedEventApplied = true;
+        }
+    }
+
+    public static class ProjectAggregateRepository
+            extends AggregateRepository<ProjectId, ProjectAggregate> {
+        public ProjectAggregateRepository() {
+            super();
+        }
+    }
+
+    @SuppressWarnings("UnusedParameters") // It is intended in this empty handler class.
+    public static class TestEventSubscriber extends EventSubscriber {
+
+        private Message handledEvent;
+
+        @Subscribe
+        public void on(ProjectCreated event, EventContext context) {
+            this.handledEvent = event;
+        }
+
+        @Subscribe
+        public void on(TaskAdded event, EventContext context) {
+        }
+
+        @Subscribe
+        public void on(ProjectStarted event, EventContext context) {
+        }
+
+        public Message getHandledEvent() {
+            return handledEvent;
+        }
+    }
+
+    public static class SecretProjectAggregate
+            extends Aggregate<String, SecretProject, SecretProjectValidatingBuilder> {
+        private SecretProjectAggregate(String id) {
+            super(id);
+        }
+
+        @Assign
+        public List<ProjectStarted> handle(StartProject cmd, CommandContext ctx) {
+            return Lists.newArrayList();
+        }
+    }
+
+    public static class SecretProjectRepository
+            extends AggregateRepository<String, SecretProjectAggregate> {
+        public SecretProjectRepository() {
+            super();
+        }
+    }
+
+    public static class ProjectProcessManager
+            extends ProcessManager<ProjectId, Empty, EmptyValidatingBuilder> {
+
+        // a ProcessManager constructor must be public because it is used via reflection
+        @SuppressWarnings("PublicConstructorInNonPublicClass")
+        public ProjectProcessManager(ProjectId id) {
+            super(id);
+        }
+
+        @Assign
+        @SuppressWarnings({"UnusedParameters", "unused"}) // OK for test method
+        public CommandRouted handle(CreateProject command, CommandContext ctx) {
+            return CommandRouted.getDefaultInstance();
+        }
+
+        @SuppressWarnings("UnusedParameters") // OK for test method
+        @Subscribe
+        public void on(ProjectCreated event, EventContext ctx) {
+            // Do nothing, just watch.
+        }
+    }
+
+    public static class ProjectPmRepo
+            extends ProcessManagerRepository<ProjectId, ProjectProcessManager, Empty> {
+
+        public ProjectPmRepo() {
+            super();
+        }
+    }
+
+    public static class ProjectReport
+            extends Projection<ProjectId, Empty, EmptyValidatingBuilder> {
+
+        @SuppressWarnings("PublicConstructorInNonPublicClass")
```

Comment:
The class is `public` now. 

In general we should eventually run IDEA's own inspections. Among other things it points to the redundant warning suppressions.